projects/plugins/boost/app/modules/optimizations/page-cache/Page_Cache_Setup.php

```diff
@@ -14,41 +15,87 @@ class Page_Cache_Setup {
 	 * @return bool|\WP_Error
 	 */
 	public static function run_setup() {
-		$steps = array(
+		// Steps that are only for cache system verification. They don't change anything.
+		$verification_steps = array(
 			'verify_wp_content_writable',
 			'verify_permalink_setting',
+		);
+
+		foreach ( $verification_steps as $step ) {
+			$result = self::run_step( $step );
+
+			if ( is_wp_error( $result ) ) {
+				return $result;
+			}
+		}
```

Comment:
🤔 This isn't a blocker, so if you really prefer it like this it doesn't have to change, but:

Wouldn't it be simpler to have all setup steps -- whether verification or setup -- always return the same things, so we only need one loop?

Reply:
I am on the fence about it, but leaning towards having them separate. This is because for the steps that can mutate something their return values can be:
- `true` - when something changed
- `false` - nothing changed but no issues
- `WP_Error` - if it failed

If we apply this same format for the verification steps, it would return `false|WP_Error`. It kind of looks weird to return `false` for success while returning true for nothing.

For mutable steps, I think it makes sense to return `bool|WP_Error` because we are checking the return value to check anything changed. Let me know what you think.

Reply:
I agree that two loops look a bit messy, but because they have a clear distinction between, I actually think keeping them separate is a step in the right direction. I'm not reviewing anything here, but just leaving my vote for doing the 2 loop approach.